File: Python/leetcode_206_ReverseLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prevNode = None
        curr = head

        while curr:
            next = curr.next
            curr.next = prevNode
            prevNode = curr
            curr = next
        return prevNode

File: Scala/leetcode_206_ReverseLinkedList.scala
/**
 * Definition for singly-linked list.
 * class ListNode(_x: Int = 0, _next: ListNode = null) {
 *   var next: ListNode = _next
 *   var x: Int = _x
 * }
 */
object Solution {
    def reverseList(head: ListNode): ListNode = {
        var prev: ListNode = null
        var curr: ListNode = head
        while (curr != null){
            val nextTemp=curr.next
            curr.next=prev
            prev=curr
            curr=nextTemp
        }
        prev
    }
}